### ISSUE TITLE
It unhighlights everything 

### ISSUE THREAD

**Issue by debonte** [microsoft/pylance-release#6442]
From Marketplace review:
> Ss Sh
> WTF is this extension it unhighlights everything ...confusing and unnecessary not to mention useless

**Comment by zhangxiwensjtu** [comment]
Here is an example.
It used to be like this:
![Image](https://github.com/user-attachments/assets/c3646ff0-2a3a-4545-a8e9-117ff218e36b)
but now it looks like this:
![Image](https://github.com/user-attachments/assets/a19c81fd-7c0a-4345-a21a-c21c2190cd18)

And also, in the function, there is no highlights:
![Image](https://github.com/user-attachments/assets/610ab0a4-7643-4ccf-933c-97f960a773a0)

**Comment by rchiodo** [comment]
That likely means the extension is not running. Can you include any output in the 'Python Language Server' tab?

**Comment by debonte** [comment]
Can you check if Pylance is both installed and enabled? When Pylance is unavailable, the Python extension will use Jedi as the Python language server instead of Pylance.

If Pylance is installed and enabled, please provide your "Python Language Server" log from the Output pane.

I'm able to reproduce the same behavior by disabling the Pylance extension:
![Image](https://github.com/user-attachments/assets/3678d2b6-e04c-432b-9562-4112b84f938b)

And then when I re-enable it, I see this:
![Image](https://github.com/user-attachments/assets/d26fecfe-341f-45fe-9645-0fe386c1a795)

Another way to check which language server you're using would be to mouse over `add_args`. Jedi shows this:
![Image](https://github.com/user-attachments/assets/c89b77b8-ac89-45ec-9992-b372b9acbf5a)

whereas Pylance shows this:
![Image](https://github.com/user-attachments/assets/fd4410a2-aa11-45fd-b7e4-b47e467e4b3a)

**Comment by zhangxiwensjtu** [comment]
> Python Language Server" lo

Hi, sorry for the late reply.
I have enabled Pylance:
![Image](https://github.com/user-attachments/assets/e645b3f5-fb22-4fd9-a544-47db334a602e)

And here is the Python Language Server output:
![Image](https://github.com/user-attachments/assets/df573d87-749f-48bb-b5f2-71b79d38bdc7)

When I put my mouse over functions, it shows "Loading...", it maybe some Vscode bug? No matter how long I wait, it stays "Loading..."
![Image](https://github.com/user-attachments/assets/a1bc06e2-8bd4-4b41-a225-689fa75945cd)

**Comment by debonte** [comment]
That sounds like Pylance either crashed or is hung. Is the log output that you included above the complete log? Does this only happen in certain projects? For example, if you open a new, empty folder in VS Code and create a new Python file there, does the issue still occur?

**Comment by zhangxiwensjtu** [comment]
> That sounds like Pylance either crashed or is hung. Is the log output that you included above the complete log? Does this only happen in certain projects? For example, if you open a new, empty folder in VS Code and create a new Python file there, does the issue still occur?

Yes! I tried to create a new repo and find Pylance works well in that new repo! Any clues on how to restart Pylance in certain project ? I tried to uninstall and reinstall it, but it doesn't work.

**Comment by debonte** [comment]
Can you share the project where you're seeing this?

**Comment by zhangxiwensjtu** [comment]
> Can you share the project where you're seeing this?

Sorry, I can't make the project code public.

Here are a few possible reasons I suspect for the issue:

1. I tried running the command du -h --max-depth=1 in the project directory and found that the .venv folder is quite large. I'm not sure if this affects the functioning of Pylance.
2. There's a symbolic link in my project directory that points to a model weight file of 100GB in size.
I suspect that these large files might be causing Pylance to malfunction. 
So I tried to create a .vscode/pyrightconfig.json file under the .vscode folder. The content is as following:
![Image](https://github.com/user-attachments/assets/cdc700b5-be81-4d1c-8925-4689adf4589c)

Then I restarted Python Language Server, and here is the complete log:
![Image](https://github.com/user-attachments/assets/2cdca48d-f038-48af-99dc-0a10bd7776bf)
It says:  The existing extension didn't exit within 10 seconds. New instance will start, but you might encounter issues.

I'm confused about it, since I've already added the symbolic link and .venv path to the pyrightconfig.json.

**Comment by GF-Huang** [comment]
Same issue due to the latest update.

![Image](https://github.com/user-attachments/assets/68a37853-1bc4-45fb-bd5b-adaa0670fb33)

And, Restart language server failed.

![Image](https://github.com/user-attachments/assets/4efeb0de-bc35-4ffd-81e6-1ba609208009)

**Comment by GF-Huang** [comment]
When I downgrade to `v2024.9.2`, all things works well.

**Comment by zhangxiwensjtu** [comment]
finally I solve this problem.
For me, the lack of syntax highlighting and navigation isn't due to a Pylance version issue, but rather because my VSCode workspace contains too many files. Even though I set up a pyrightconfig.json file to ignore those large folders, I discovered that the pyrightconfig.json file wasn't actually working. When I started VSCode, the Python language server still reported a timeout error when scanning files. Finally, I found that setting the exclude file paths in this specific location resolved the issue.

![Image](https://github.com/user-attachments/assets/43b7ce8e-8089-4c6b-ab26-e1e90f85a09a)
reference : https://github.com/microsoft/pylance-release/wiki/Opening-Large-Workspaces-in-VS-Code

**Comment by judej** [comment]
@zhangxiwensjtu, glad you found that the wiki was helpful. Appreciate the update.
 
@GF-Huang, would you please check if this fixes your issue too? If not, would you please create  a new issue and add the information in the report as described in [the troubleshooting guide](https://github.com/microsoft/pylance-release/blob/main/TROUBLESHOOTING.md#filing-an-issue)?

**Comment by xhluca** [comment]
This issue still happens a lot to me, when i'm ssh'd to a remote. Using the "exclude" trick above doesn't help.

Usually i would change to pre-release then reload, sometimes that fixes it, but the next time I SSH to the remote the issue arises again (everything unhighlighted).

I'm not sure what's the cause as I didn't change much from my workflow apart, this simply started happening a month or two ago.

**Comment by rchiodo** [comment]
@xhluca can you open a new issue? Generally if highlighting disappears, it's because Pylance isn't running for some reason. We'd likely need the log as described here:
https://github.com/microsoft/pylance-release/blob/main/TROUBLESHOOTING.md#filing-an-issue
